src/views/TradingCompetition/components/TeamRanks/TopTradersCard/ExpandedGridItem.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react'
+import { Text, Flex, Box } from '@pancakeswap-libs/uikit'
+import styled from 'styled-components'
+import { LeaderboardDataItem } from '../../../types'
+import { localiseTradingVolume, accountEllipsis } from '../../../helpers'
+import { LeaderboardStorm, LeaderboardFlippers, LeaderboardCakers } from '../../../svgs'
+
+const Wrapper = styled.div`
+  border-bottom: 1px solid ${({ theme }) => theme.colors.textDisabled};
+  display: grid;
+  grid-gap: 8px;
+  grid-template-columns: 1fr 4fr auto;
+
+  /* Specific styles for between 576 - 852px where the expanded wrapper shows as a three-column grid */
+  ${({ theme }) => theme.mediaQueries.sm} {
+    grid-gap: 4px;
+    grid-template-columns: auto 1fr auto;
+  }
+
+  ${({ theme }) => theme.mediaQueries.md} {
+    grid-gap: 16px;
+    grid-template-columns: 1fr 4fr auto;
+  }
+
+  svg {
+    height: 44px;
+    width: auto;
+  }
+`
+
+const RankItem = styled(Flex)`
+  margin-left: 4px;
+
+  ${({ theme }) => theme.mediaQueries.xs} {
+    margin-left: 8px;
+  }
+
+  ${({ theme }) => theme.mediaQueries.md} {
+    margin-left: 16px;
+  }
+`
+
+const VolumeAddressWrapper = styled(Box)`
+  display: grid;
+  grid-template-columns: 1fr 2fr;
+
+  /* Between 576 - 852px the expanded wrapper shows as a three-column grid and these elements are stacked */
+  ${({ theme }) => theme.mediaQueries.sm} {
+    display: flex;
+    flex-direction: column;
+  }
+
+  ${({ theme }) => theme.mediaQueries.md} {
+    display: grid;
+  }
+`
+
+const VolumeText = styled(Text)`
+  margin-right: 8px;
+
+  /* Specific style for between 576 - 852px where the expanded wrapper shows as a three-column grid */
+  ${({ theme }) => theme.mediaQueries.sm} {
+    margin-right: 0;
+  }
+
+  ${({ theme }) => theme.mediaQueries.md} {
+    margin-right: 16px;
+  }
+`
+
+const GridItem: React.FC<{ traderData?: LeaderboardDataItem }> = ({ traderData }) => {
+  const { address, volume, teamId, rank } = traderData
+
+  const icon = {
```

Comment:
why don't you use an array ?

Reply:
Did this & moved to the level above as it's shared by both components 👌